tasks/ops.ts

```diff
@@ -154,6 +131,64 @@ task("quest-add", "Adds a quest to the staked token")
         const expiry = Math.floor(Date.now() / 1000)
         const addQuestData = questManager.interface.encodeFunctionData("addQuest", [type, taskArgs.multiplier, expiry])
         console.log(`Destination ${questManagerAddress}, data: ${addQuestData}`)
-        // const tx = await questManager.addQuest(type, taskArgs.multiplier, expiry)
-        // await logTxDetails(tx, `Add ${taskArgs.type} quest with ${taskArgs.multiplier} multiplier`)
+        const tx = await questManager.addQuest(type, taskArgs.multiplier, expiry)
+        await logTxDetails(tx, `Add ${taskArgs.type} quest with ${taskArgs.multiplier} multiplier`)
+    })
+
+task("quest-complete-queue", "Completes all user quests in the quests queue")
+    .addOptionalParam("speed", "Defender Relayer speed param: 'safeLow' | 'average' | 'fast' | 'fastest'", "fast", types.string)
+    .addParam("signerKey", "Signer API key", undefined, types.string, false)
+    .addParam("signerSecret", "Signer API secret", undefined, types.string, false)
+    .setAction(async (taskArgs, hre) => {
+        const opsSigner = await getSigner(hre, taskArgs.speed)
+        const chain = getChain(hre)
+        const questManagerAddress = await resolveAddress("QuestManager", chain)
+        const questManager = QuestManager__factory.connect(questManagerAddress, opsSigner)
+
+        // get users who have completed quests from the queue
+        const response = await axios.post("https://europe-west1-mstable-questbook.cloudfunctions.net/questbook", {
+            query: `query { queue { userId ethereumId } }`,
+        })
+        const { queue } = response?.data?.data
+        if (!queue) {
+            console.log(response?.data)
+            throw Error(`Failed to get quests from queue`)
+        }
+        if (queue.length === 0) {
+            console.error(`No user completed quests`)
+            process.exit(0)
+        }
+        // filter users to just the migration quest
+        const migrationQuestId = 0
+        const completedMigrationQuests = queue.filter((quest) => quest.ethereumId === migrationQuestId)
+        const completedMigrationUsers = completedMigrationQuests.map((quest) => quest.userId)
+
+        // Need to filter out any users that completed the quest themselves
+        const hasCompletedPromises = completedMigrationUsers.map((user) => questManager.hasCompleted(user, migrationQuestId))
+        const hasCompleted = await Promise.all(hasCompletedPromises)
```

Comment:
Should add another line here for actually pinging the API to check for completion, and filter out the users who do not meet the criteria (as a secondary validation check)

Reply:
Good idea. I'll add

Reply:
I've added a check against Quest Book's `user` query for each user that is claiming quest completion.